amethyst_utils/src/removal.rs

```diff
@@ -0,0 +1,73 @@
+use amethyst_assets::{PrefabData, PrefabError};
+use amethyst_core::specs::{Component, DenseVecStorage, Entity, Join, ReadStorage, WriteStorage};
+use amethyst_core::specs::world::EntitiesRes;
+use std::fmt::Debug;
+use std::result::Result;
+
+/// A marker `Component` used to remove entities and clean up your scene.
+/// The generic parameter `I` is the type of id you want to use.
+/// Generally an int or an enum.
+#[derive(Debug, Clone)]
+pub struct Removal<I> 
```

Comment:
I don't think this should be called `Removal` as you could use it for other purposes that need to identify all entities made for a specific scene. Maybe something like `SceneTag` instead?

Reply:
Its better to use other types to identify entities. If you re-use this, all the identifiers will be in a giant list. If you use different types for identifications, it makes smaller list because they are divided into Storage<T> instead of a single (Storage<Removal<i32>> -> runtime match what you need)